src/implementations/svd.jl

```diff
@@ -66,6 +94,18 @@ function initialize_output(::typeof(svd_trunc!), A::AbstractMatrix, alg::Truncat
     return initialize_output(svd_compact!, A, alg.alg)
 end
 
+function initialize_output(::typeof(svd_full!), A::Diagonal, ::DiagonalAlgorithm)
+    TA = eltype(A)
+    TUV = Base.promote_op(sign_safe, TA)
+    return similar(A, TUV, size(A)), similar(A, real(TA)), similar(A, TUV, size(A))
```

Comment:
If you want `Diagonal` `U` and `Vᴴ` there shouldn't be any `size` arguments in `similar`.

Reply:
See other comment, I specifically don't want `Diagonal`s here.